.github/workflows/deploy.yaml

```diff
@@ -56,11 +56,66 @@ jobs:
           yarn util:link
           yarn build
 
+  publish_npm:
+    name: Publish to NPM
+    runs-on: ubuntu-latest
+    # we must wait for the possible commit changes from the smart contract
+    # deployment for a full version
+    needs: [build, deploy_smart_contracts]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          # we use a custom PAT to allow our workflows to run on this commit,
```

Comment:
can you explain?

Reply:
by default the checkout action uses a session token which prevents workflows to trigger when new commits are pushed as a circuit breaker. But we want to run workflows on that new version commit, so we need to use our own token.

Reply:
obsolete